src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -697,7 +697,6 @@ class JSONReader
 
             const bool isPartial = orjson::GetJsonValueByKey<bool>(rRefKinBodyInfo, "__isPartial__", true);
             RAVELOG_VERBOSE_FORMAT("Deserializing rRefKinBodyInfo=%s into insertIndex=%d, isPartial=%d", orjson::DumpJson(rRefKinBodyInfo)%insertIndex%isPartial);
-            pNewKinBodyInfo->Reset();
```

Comment:
mhshp 6/7 and 7/7 failed. 

when we have body1 -> body2 -> body3, this line will wipe out data in body3, potentially making kinbody incomplete.

Pipeline #571837

Reply:
@LLipter can you try again with my latest change?